### DISCUSSION TITLE
What is the behavior of 'ckpt_path' when passed into trainer.fit()?

### DISCUSSION THREAD

**Discussion by JLenzy** [Lightning-AI/pytorch-lightning#20499] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
I am trying to identify what elements of my saved pl checkpoint are going to be resume from checkpoint, and what needs to be manually re-specified. For example, when I did my initial training run, let's say I did:

```
trainer = pl.Trainer(args, max_steps=500)
```
and now I am trying to resume:
```
my_model = Model.load_from_checkpoint(ckpt_path)
...
trainer = pl.Trainer(my_model)
trainer.fit(ckpt_path=ckpt_path)
```

So I've now passed in ckpt twice, to load the model (which I need to do prior to the trainer, for logging purposes) and now to the trainer itself. Will things like my config's learning rate, max number of steps, number of val batches, etc. still be saved and utilized from the previous training run? Or do I need to pass in the original config file to re-duplicate it?
